src/common.rs

```diff
@@ -156,7 +156,26 @@ impl Frame<'static> {
         
         
     }
-    
+
+    /// Creates a frame from a palette and indexed pixels
+    pub fn from_palette_pixels(width: u16, height: u16, palette: &[u8], pixels: &mut [u8]) -> Frame<'static> {
```

Comment:
Just change the signature to 

```rust
pub fn from_palette_pixels(width: u16, height: u16, pixels: &mut [u8], palette: &[u8], transparent: Option<u8>) -> Frame
```

such that you can both set the transparency properly and remove the clones.